include/rive/playable.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef _RIVE_PLAYABLE_HPP_
+#define _RIVE_PLAYABLE_HPP_
+
+#include "rive/animation/loop.hpp"
+#include "rive/math/aabb.hpp"
+#include "rive/math/vec2d.hpp"
+#include <string>
+
+namespace rive {
+    class ArtboardInstance;
+    class Renderer;
+
+    class SMIInput;
+    class SMIBool;
+    class SMINumber;
+    class SMITrigger;
+
+    class Playable {    
+    protected:
+        ArtboardInstance* m_ArtboardInstance;
+
+        Playable(ArtboardInstance*);
+
+    public:
+        virtual ~Playable() {}
+    
+        float width() const;
+        float height() const;
+        AABB bounds() const { return {0, 0, this->width(), this->height()}; }
+
+        virtual std::string name() const = 0;
+
+        virtual Loop loop() const = 0;
```

Comment:
Why do we expose this? A StateMachine will never have this. I wonder if certain "concrete specific playable" properties should be exposed differently, like querying which animations/stateMachines are in the playable and exposing simple interfaces for each (instead of our deep/actual ones).